src/audio/codec_adapter/codec_adapter.c

```diff
@@ -404,19 +396,21 @@ int codec_adapter_copy(struct comp_dev *dev)
 		goto db_verify;
 	}
 
-	/* Process only we have enough free data in the local
-	 * buffer. If we don't have enough free space process()
-	 * will override the data in the local buffer
-	 */
-	if (local_buff->stream.free < get_output_bytes(dev))
-		goto db_verify;
-
 	buffer_stream_invalidate(source, codec_buff_size);
 	codec_adapter_copy_from_source_to_lib(&source->stream, &codec->cpd,
 					      codec_buff_size);
 	codec->cpd.avail = codec_buff_size;
 	ret = codec_process(dev);
 	if (ret) {
+		if (ret == -ENOSPC) {
+			/* copy the data back to the source buffer as it was not used */
+			codec_adapter_copy_from_lib_to_sink(&codec->cpd, &source->stream,
```

Comment:
maybe rename it to `codec_adapter_copy_from_lib_to_stream()` to make it less confusing?

Reply:
I'm going to save all the renaming to later because the whole codec_adapter is going to rename with module_interface and the lib will have to go at that point too.